meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -394,4 +414,69 @@ public byte[] serialize(T data, String path) {
   public T deserialize(byte[] bytes, String path) {
     return _zkClient.deserialize(bytes, path);
   }
+
+  /**
+   * A clean up method called when connect state change or MetaClient is closing.
+   * @param cancel If we want to cancel the reconnect monitor thread.
+   * @param close If we want to close ZkClient.
+   */
+  private void cleanUpAndClose(boolean cancel, boolean close) {
```

Comment:
Nit: It is already well explained in your comment, but it would be better if the method signature is something like `private void cleanUpAndClose(boolean cancelMonitor, boolean closeZKClient)`.

Reply:
TFTR. Good suggestion. Will update.